File: solutions/java/darts/1/src/main/java/Darts.java
class Darts {
    int score(double xOfDart, double yOfDart) {
        double distance = Math.sqrt(xOfDart * xOfDart + yOfDart * yOfDart);

        if (distance > 10) {
            return 0;
        }

        if (distance <= 10 && distance > 5) {
            return 1;
        }

        if (distance <= 5 && distance > 1) {
            return 5;
        }

        return 10;
    }

}

File: solutions/kotlin/darts/2/src/main/kotlin/Darts.kt
import kotlin.math.sqrt

object Darts {
    fun score(x: Number, y: Number): Int {
        val d = distance(x.toDouble(), y.toDouble());

        return when (d) {
            in 0.0..1.0 -> 10
            in 1.0..5.0 -> 5
            in 5.0..10.0 -> 1
            else -> 0
        }
    }

    fun distance (x: Double, y: Double): Double {
        val a = x * x;
        val b = y * y;

        return sqrt((a + b).toDouble())
    }
}

    
